galaxy_ng/app/api/v1/viewsets.py

```diff
@@ -0,0 +1,358 @@
+import datetime
+import logging
+
+from django.conf import settings
+from django.db.models import Q
+from django.http import HttpResponse
+from rest_framework.response import Response
+from rest_framework import viewsets
+
+from rest_framework.authentication import BasicAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import AllowAny
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.exceptions import ValidationError
+
+from pulpcore.plugin.models import Task
+from pulpcore.plugin.tasking import dispatch
+
+from galaxy_ng.app.api.v1.tasks import legacy_role_import
+from galaxy_ng.app.api.v1.tasks import legacy_sync_from_upstream
+from galaxy_ng.app.api.v1.models import LegacyNamespace
+from galaxy_ng.app.api.v1.models import LegacyRole
+from galaxy_ng.app.api.v1.serializers import LegacyRoleSerializer
+from galaxy_ng.app.api.v1.serializers import LegacyRoleContentSerializer
+from galaxy_ng.app.api.v1.serializers import LegacyRoleVersionsSerializer
+from galaxy_ng.app.api.v1.serializers import LegacyUserSerializer
+
+
+logger = logging.getLogger(__name__)
+
+
+class LegacyUserSetPagination(PageNumberPagination):
+    page_size = 100
+    page_size_query_param = 'page_size'
+    max_page_size = 1000
+
+
+class LegacyUserViewSet(viewsets.ModelViewSet):
+    queryset = LegacyNamespace.objects.all().order_by('name')
+    serializer = LegacyUserSerializer
+    serializer_class = LegacyUserSerializer
+    permission_classes = [AllowAny]
```

Comment:
@newswangerd is it alright if we address the v1 RBAC edits in a subsequent PR that we work after this is merged?

Reply:
I want RBAC to be consistent. If he handle permission checking in multiple places, it's going to get really hard to audit the security. It's not super hard to add the access policies for this, so I would like to just do it now rather than risking it falling through the cracks and creating security issues later.

Reply:
I was thinking our `DEPLOYMENT_MODE=community` deployment from `master` progress may be blocked by this PR not being merged, but at least right now it is not, :+1:

Reply:
fixed.